InputState input_state
InputState[] negative_states
---
ActivationState activation_state